### ISSUE TITLE
Search features by raw OSM tag names

### ISSUE THREAD

**Issue by pastk** [organicmaps/organicmaps#8348]
Its not uncommon for OSM mappers and power users to try search features via their OSM tag e.g. `amusement_arcade`, `fitness_station`, etc.

Its quite logical and useful for mappers, but it doesn't work atm.

This kind of search should skip the grouping set in categories.txt and search for a single feature type.
The "tag keyword" could be derived from internal OM type name (though it won't work for 100% cases because sometimes our internal names are intentionally different from OSM).

I guess the downside will be that the search index will grow in size, right @vng ?

**Comment by RedAuburn** [comment]
i'm not sure that it's worth doing this. Speaking _as a mapper_, OM targets normal users - not mappers. And needing to resort to searching by tag is an indication that OM search should instead handle naturally searching types better.

**Comment by eisams** [comment]
> OM search should instead handle naturally searching types better.

Yes, or let you browse subcategories. E.g., you select "Hotels" and you can subsequently filter by the types shown on the map (Hostel, Hotel, Guest House, Wilderness Hut etc.,)

**Comment by Misalf-git** [comment]
Translation of OSM tags/features used in OM can be ambiguous, confusing or incorrect.

**Comment by RedAuburn** [comment]
> Translation of OSM tags/features used in OM can be ambiguous, confusing or incorrect.

then we should fix that :)

**Comment by pastk** [comment]
> i'm not sure that it's worth doing this. Speaking _as a mapper_, OM targets normal users - not mappers. And needing to resort to searching by tag is an indication that OM search should instead handle naturally searching types better.

Of course its not a replacement for "natural" search.
Its a complimentary feature for experienced mappers.

**Comment by dadofsambonzuki** [comment]
Having this as an advance feature with appropriate caveats would be great imo.

I know of users executing Overpass queries for given tags and then manually importing them as KZMs. They have to do this periodically obv, which is a pita.

**Comment by eFFecTM** [comment]
I like to see this feature too, I would like to search on tags such as barrier = stile (as crazy that might sound).
